### PR TITLE
[train] reference custom resources in ScalingConfig.resources_per_worker

### PR DESCRIPTION
**Pull request by matthewdeng**
## Why are these changes needed?

Make it clear that you can specify Ray Core custom resources here.

image

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
Does it support memory without custom resources or only cpu and gpu?

**Review by justinvyu — COMMENTED** [review]
Add custom resources to the example below:

```python
from ray.train import ScalingConfig

scaling_config = ScalingConfig(
    # Number of distributed workers.
    num_workers=2,
    # Turn on/off GPU.
    use_gpu=True,
    # Assign extra CPU/GPU/custom resources per worker.
    resources_per_worker={"GPU": 1, "CPU": 1, "memory": 1e9, "custom": 1.0},
    # Try to schedule workers on different nodes.
    placement_strategy="SPREAD",
)
```

Also:
1. Let's remove this `trainer_resources` section in the docs, or just keep the section talking about setting it to 0. https://docs.ray.io/en/latest/train/user-guides/using-gpus.html#trainer-resources
2. Link the scaling gpu resource configuration guide at the top of the `ScalingConfig` docstring.
3. Update the `resources_per_worker` section to not emphasize setting multiple GPUs. We already have another section about this, and it's not the recommended pattern.

File: python/ray/air/config.py [hunk 1]
```diff
@@ -120,8 +122,9 @@ class ScalingConfig:
             argument.
         resources_per_worker: If specified, the resources
             defined in this Dict is reserved for each worker.
-            Define the ``"CPU"`` and ``"GPU"`` keys (case-sensitive) to
-            override the number of CPU or GPUs used by each worker.
+            Define the ``"CPU"`` key (case-sensitive) to
+            override the number of CPUs used by each worker.
+            This can also be used to request :ref:`custom resources <custom-resources>`.
```
Comment (hongpeng-guo) [inline python/ray/air/config.py:127]: Does this link to Ray Core's [Custom Resources](https://docs.ray.io/en/latest/ray-core/scheduling/resources.html#custom-resources) Page?

**Review by hongpeng-guo — APPROVED** [review]
LGTM

File: python/ray/air/config.py (hunk 1 above)
Comment (hongpeng-guo) [reply python/ray/air/config.py:127]: Ohh yeah, got it from the PR description.

**Pull request merged**